src/protocol/vm/tycho_decoder.rs

```diff
@@ -113,15 +117,15 @@ impl TryFromWithBlock<ComponentWithState> for VMPoolState<PreCachedDB> {
             .map(H160::from_bytes)
             .collect();
 
-        let adapter_file_path = format!(
-            "src/protocol/vm/assets/{}",
-            to_adapter_file_name(
-                snapshot
-                    .component
-                    .protocol_system
-                    .as_str(),
-            )
-        );
+        let protocol_name = snapshot
+            .component
+            .protocol_system
+            .strip_prefix("vm:")
```

Comment:
What if we have a VM and a native implementation?

Reply:
True! I'll make the stripping optional.

Reply:
On that note, is there a reason we still need the 'vm:' prefix for these protocol names in tycho?

Reply:
I guess if we have both a VM and native implementation we could distinguish them like this.